app/src/androidTest/java/com/duckduckgo/app/browser/downloader/DataUriParserTest.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2018 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.downloader
+
+import com.duckduckgo.app.browser.downloader.DataUriParser.ParseResult.Invalid
+import com.duckduckgo.app.browser.downloader.DataUriParser.ParseResult.ParsedDataUri
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+
+class DataUriParserTest {
+
+    private lateinit var testee: DataUriParser
+
+    @Before
+    fun setup() {
+        testee = DataUriParser()
+    }
+
+    @Test
+    fun whenMimeTypeProvidedAsImagePngThenPngSuffixGenerated() {
+        val parsed = testee.generate("data:image/png;base64,AAAA") as ParsedDataUri
+        assertEquals("png", parsed.filename.fileType)
+    }
+
+    @Test
+    fun whenMimeTypeProvidedAsImageJpegThenJpegSuffixGenerated() {
+        val parsed = testee.generate("data:image/jpeg;base64,AAAA") as ParsedDataUri
+        assertEquals("jpeg", parsed.filename.fileType)
+    }
+
+    @Test
+    fun whenMimeTypeProvidedAsWildcardImageTypeThenDefaultJpgSuffixGenerated() {
+        val parsed = testee.generate("data:image/*;base64,AAAA") as ParsedDataUri
+        assertEquals("jpg", parsed.filename.fileType)
+    }
```

Comment:
Given that the user must have initiated this from an actual image I guess we're safe to assume it really is an image, but if we don't know what type then this feels as though we could easy be getting the image type wrong as often as we get it right.  

For images of unknown type we could use `BitmapFactory` to decode them, then use `Bitmap.compress` to re-encode them in to a known type (e.g. JPG).  One downside I see with this is animated images would likely just end up being a single frame, however I think that is better than just assuming they are a JPG.

Reply:
I'm just going to defer all of this logic to `getExtensionFromMimeType` instead of us trying to be clever.